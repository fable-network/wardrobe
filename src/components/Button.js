import React from 'react';
import PropTypes from 'prop-types';
import styled  from 'styled-components';

const StyledButton = styled.button`
  display: inline-block;
  padding: 5px 20px;
  background-color: ${props => (props.disabled ? '#cde0ed' : '#5f9dc7')}; // theme
  font-family: 'Avenir Next', sans-serif; // theme
  font-weight: normal;
  color: #fff; // theme
  font-size: ${props => Button.sizes[props.size]};
  // Reset default button styles
  border: 0;

  &:not([disabled]):hover {
    background-color: ${props => props.theme.skyBlue}; // theme
    box-shadow: 0 2px 4px 0 rgba(49, 50, 51, 0.1); // theme?
  }
`;

/**
 * An element of a graphical user interface which a user can select to perform a particular action.
 */
const Button = ({ size, disabled, children }) => (
  <StyledButton size={size} disabled={disabled}>
    {children}
  </StyledButton>
);

Button.propTypes = {
  /**
   * Button label.
   */
  children: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['normal']),
  disabled: PropTypes.bool,
  theme: PropTypes.object.isRequired,
};

Button.defaultProps = {
  size: 'normal',
  disabled: false,
};

Button.sizes = {
  normal: '18px',
};

/** @component */
export default Button;
