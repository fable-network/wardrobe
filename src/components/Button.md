Basic button:

```jsx
<Button>Push Me</Button>
```

Disabled button:

```jsx
<Button disabled>Push Me</Button>
```
