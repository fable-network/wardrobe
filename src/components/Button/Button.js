import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { paddingHorizontal, paddingVertical } from '../../helpers/styled';

const cssDefaultDisabled = css`
  &:disabled,
  &:hover:disabled,
  [data-whatintent='keyboard'] &:focus:disabled {
    color: ${p => p.theme.grey03};
    background-color: ${p => p.theme.grey06};
    border-color: ${p => p.theme.grey06};
  }
`;

const cssPrimary = css`
  color: ${p => p.theme.white};
  background-color: ${p => p.theme.primary};
  border-color: ${p => p.theme.primary};
  border-radius: ${p => p.theme.borderRadius};

  &:hover,
  [data-whatintent='keyboard'] &:focus {
    background-color: ${p => p.theme.primaryActive};
    border-color: ${p => p.theme.primaryActive};
  }

  ${cssDefaultDisabled};
`;

const cssSecondary = css`
  color: ${p => p.theme.primary};
  background-color: ${p => p.theme.white};
  border-color: ${p => p.theme.primary};
  border-radius: ${p => p.theme.borderRadius};

  &:hover,
  [data-whatintent='keyboard'] &:focus {
    color: ${p => p.theme.primaryActive};
    border-color: ${p => p.theme.primaryActive};
  }

  ${cssDefaultDisabled};
`;

const cssDanger = css`
  color: ${p => p.theme.white};
  background-color: ${p => p.theme.danger};
  border-color: ${p => p.theme.danger};
  border-radius: ${p => p.theme.borderRadius};

  ${cssDefaultDisabled};
`;

const cssSuccess = css`
  color: ${p => p.theme.white};
  background-color: ${p => p.theme.success};
  border-color: ${p => p.theme.success};
  border-radius: ${p => p.theme.borderRadius};

  ${cssDefaultDisabled};
`;

const cssWarning = css`
  color: ${p => p.theme.white};
  background-color: ${p => p.theme.warning};
  border-color: ${p => p.theme.warning};
  border-radius: ${p => p.theme.borderRadius};

  ${cssDefaultDisabled};
`;

const cssLink = css`
  background-color: transparent;
  &,
  &:link,
  &:visited {
    text-decoration: none;
    color: ${p => p.theme.primary};
    box-shadow: ${p => p.theme.noShadow};
  }

  [data-whatintent='keyboard'] &:focus {
    background-color: ${p => p.theme.grey04};
    text-decoration: underline;
  }

  &:active,
  &:hover {
    box-shadow: ${p => p.theme.noShadow};
    text-decoration: underline;
    color: ${p => p.theme.primaryActive};
  }

  &:disabled {
    &,
    &:link,
    &:active,
    &:hover,
    &:visited,
    [data-whatintent='keyboard'] &:focus {
      cursor: not-allowed;
      background: none;
      text-decoration: none;
      color: ${p => p.theme.grey04};
    }
  }
`;

const cssLarge = css`
  ${p => paddingHorizontal(`calc(${p.theme.paddingHorizontalLarge} - 1px)`)};
  ${p => paddingVertical(`calc(${p.theme.paddingVerticalLarge} - 1px)`)};
`;

const ButtonInner = ({ size, appearance, renderAs, children, ...otherProps }) =>
  React.createElement(renderAs, otherProps, children);

ButtonInner.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.oneOf(['normal', 'large']),
  appearance: PropTypes.oneOf(['primary', 'secondary', 'link', 'success', 'danger', 'warning']),
  renderAs: PropTypes.string,
};

ButtonInner.defaultProps = {
  size: 'normal',
  appearance: 'secondary',
  renderAs: 'button',
};

/**
 * An element of a graphical user interface which a user can select to perform a particular action.
 */
const Button = styled(ButtonInner)`
  display: inline-flex;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  font-size: ${p => p.theme.fontSizeSmall};
  line-height: ${p => p.theme.lineHeightControlSmall};
  ${p => paddingHorizontal(`calc(${p.theme.paddingHorizontalBase} - 1px)`)};
  ${p => paddingVertical(`calc(${p.theme.paddingVerticalBase} - 1px)`)};
  border: solid 1px transparent;
  transition: border-color 150ms linear, background-color 150ms linear, box-shadow 150ms linear;
  cursor: pointer;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  font-family: inherit;
  font-weight: normal;

  &:hover {
    box-shadow: ${p => p.theme.shadow};
    transition: border-color 150ms linear, background-color 150ms linear, box-shadow 150ms linear;
  }

  &:disabled {
    cursor: not-allowed;
  }

  &:hover:disabled {
    box-shadow: ${p => p.theme.noShadow};
  }

  ${p => p.appearance === 'primary' && cssPrimary};
  ${p => p.appearance === 'secondary' && cssSecondary};
  ${p => p.appearance === 'danger' && cssDanger};
  ${p => p.appearance === 'success' && cssSuccess};
  ${p => p.appearance === 'warning' && cssWarning};
  ${p => p.appearance === 'link' && cssLink};
  ${p => p.size === 'large' && cssLarge};
`;

Button.propTypes = {
  /** Button label */
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  size: PropTypes.oneOf(['normal', 'large']),
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
  appearance: PropTypes.oneOf(['primary', 'secondary', 'link', 'success', 'danger', 'warning']),
  /** Render to DOM as a given tag. Should be a valid HTML tag. */
  renderAs: PropTypes.string,
};

Button.defaultProps = {
  size: 'normal',
  disabled: false,
  appearance: 'secondary',
  renderAs: 'button',
};

/** @component */
export default Button;
