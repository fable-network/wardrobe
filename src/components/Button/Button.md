Primary button:

```jsx
<Button appearance="primary">Push Me</Button>
```

Large button:

```jsx
<Button appearance="primary" size="large">
  Push Me
</Button>
```

Disabled primary button:

```jsx
<Button appearance="primary" disabled>
  Don't Push Me
</Button>
```

Secondary button:

```jsx
<Button appearance="secondary">Push Me</Button>
```

Disabled secondary button:

```jsx
<Button appearance="secondary" disabled>
  Don't Push Me
</Button>
```

Danger Button:

```jsx
<Button appearance="danger">Dangerous action</Button>
```

Success Button:

```jsx
<Button appearance="success">Push Me</Button>
```

Warning Button:

```jsx
<Button appearance="warning">Push Me</Button>
```

Link button:

```jsx
<Button appearance="link">Push Me</Button>
```

Disabled link button:

```jsx
<Button appearance="link" disabled>
  Don't Push Me
</Button>
```

Multiline button:

```jsx
<div>
  <Button>
    Push Me<br />Touch me
  </Button>
</div>
```

On a grid:

```jsx
const EightPxGrid = require('../../styleguide/EightPxGrid.js').default;
const styled = require('styled-components').default;
const Wrapper = styled.div`
  display: flex;
  flex-flow: row nowrap;
  align-items: flex-start;
  > * + * {
    margin-left: 24px;
  }
`;
const ButtonCustom = styled(Button)`
  font-size: 2rem;
`;
<EightPxGrid>
  <Wrapper>
    <Button>Normal</Button>
    <Button size="large">Large</Button>
    <Button>
      Push Me<br />Touch me
    </Button>
    <ButtonCustom>Custom</ButtonCustom>
  </Wrapper>
</EightPxGrid>;
```

You can also render Button as a given HTML tag, e.g. if you want it to look like a button but wrap it with a react-router's `Link` component:

```jsx
<Button appearance="primary" renderAs="span">
  I'm not a button
</Button>
```
