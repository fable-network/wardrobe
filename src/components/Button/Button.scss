//
// Button
//
@import '../../globals/scss/colors';
@import '../../globals/scss/typography';
@import '../../globals/scss/variables';
@import '../../globals/scss/import-once';

// Import this only once
@include exports('button') {

  .#{$prefix}--btn {
    display: inline-block;
    padding: 5px 20px;
    font-family: inherit;
    font-weight: 300;
    font-size: $button-font-size;
    border: 0;
    transition: border-color 150ms linear, background-color 150ms linear, box-shadow 150ms linear;

    &:hover {
      box-shadow: 0 2px 4px 0 rgba(49, 50, 51, 0.2);
      transition: border-color 150ms linear, background-color 150ms linear, box-shadow 150ms linear;
    }

    &:disabled {
      cursor: not-allowed;
    }

    &:hover:disabled {
      background: lighten($color__skyBlue, 30%);
      box-shadow: none;
    }
  }

  .#{$prefix}--btn--primary {
    color: $color__pearlWhite;
    background-color: $color__skyBlue;
    border: 1px solid $color__skyBlue; // to get same size as 'secondary' styled button.

    &:hover {
      background-color: darken($color__skyBlue, 20%);
      border-color: darken($color__skyBlue, 20%);
    }

    &:disabled,
    &:hover:disabled {
      background-color: lighten($color__skyBlue, 30%);
      border-color: lighten($color__skyBlue, 30%);
    }
  }

  .#{$prefix}--btn--secondary {
    color: $color__ravenBlack;
    background-color: $color__pureWhite;
    border: 1px solid $color__ravenBlack;

    &:disabled {
      color: lighten($color__ravenBlack, 50%);
      border-color: lighten($color__ravenBlack, 50%);
    }

    &:hover:disabled {
      background: $color__pureWhite;
      color: lighten($color__ravenBlack, 50%);
      border-color: lighten($color__ravenBlack, 50%);
    }
  }

  .#{$prefix}--btn--sm {
    font-size: 0.875rem;
    padding: 3px 10px;
  }

}
