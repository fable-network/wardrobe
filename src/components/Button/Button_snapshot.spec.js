/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = shallow(<Button appearance="primary">Push Me</Button>);
      expect(wrapper1).toMatchSnapshot();
        
      const wrapper2 = shallow(<Button appearance="primary" disabled>Don't Push Me</Button>);
      expect(wrapper2).toMatchSnapshot();
        
      const wrapper3 = shallow(<Button appearance="secondary">Push Me</Button>);
      expect(wrapper3).toMatchSnapshot();
        
      const wrapper4 = shallow(<Button appearance="secondary" disabled>Don't Push Me</Button>);
      expect(wrapper4).toMatchSnapshot();
        
      const wrapper5 = shallow(<Button size="small">Push Me</Button>);
      expect(wrapper5).toMatchSnapshot();
        
    });
  });
});
