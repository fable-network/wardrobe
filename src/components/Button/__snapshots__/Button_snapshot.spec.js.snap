// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="primary"
    disabled={false}
    renderAs="button"
    size="normal"
  >
    Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "primary",
      "children": "Push Me",
      "className": "Button-hYXUXp fzSAgN",
      "disabled": false,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "primary",
        "children": "Push Me",
        "className": "Button-hYXUXp fzSAgN",
        "disabled": false,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="primary"
    disabled={false}
    renderAs="button"
    size="large"
  >
    Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "primary",
      "children": "Push Me",
      "className": "Button-hYXUXp dWPhUy",
      "disabled": false,
      "renderAs": "button",
      "size": "large",
    },
    "ref": null,
    "rendered": "Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "primary",
        "children": "Push Me",
        "className": "Button-hYXUXp dWPhUy",
        "disabled": false,
        "renderAs": "button",
        "size": "large",
      },
      "ref": null,
      "rendered": "Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="primary"
    disabled={true}
    renderAs="button"
    size="normal"
  >
    Don't Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "primary",
      "children": "Don't Push Me",
      "className": "Button-hYXUXp fzSAgN",
      "disabled": true,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Don't Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "primary",
        "children": "Don't Push Me",
        "className": "Button-hYXUXp fzSAgN",
        "disabled": true,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Don't Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="secondary"
    disabled={false}
    renderAs="button"
    size="normal"
  >
    Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "secondary",
      "children": "Push Me",
      "className": "Button-hYXUXp hgwSQR",
      "disabled": false,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "secondary",
        "children": "Push Me",
        "className": "Button-hYXUXp hgwSQR",
        "disabled": false,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 5`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="secondary"
    disabled={true}
    renderAs="button"
    size="normal"
  >
    Don't Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "secondary",
      "children": "Don't Push Me",
      "className": "Button-hYXUXp hgwSQR",
      "disabled": true,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Don't Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "secondary",
        "children": "Don't Push Me",
        "className": "Button-hYXUXp hgwSQR",
        "disabled": true,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Don't Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 6`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="danger"
    disabled={false}
    renderAs="button"
    size="normal"
  >
    Dangerous action
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "danger",
      "children": "Dangerous action",
      "className": "Button-hYXUXp csvHuG",
      "disabled": false,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Dangerous action",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "danger",
        "children": "Dangerous action",
        "className": "Button-hYXUXp csvHuG",
        "disabled": false,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Dangerous action",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 7`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="success"
    disabled={false}
    renderAs="button"
    size="normal"
  >
    Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "success",
      "children": "Push Me",
      "className": "Button-hYXUXp fBofzX",
      "disabled": false,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "success",
        "children": "Push Me",
        "className": "Button-hYXUXp fBofzX",
        "disabled": false,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 8`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="warning"
    disabled={false}
    renderAs="button"
    size="normal"
  >
    Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "warning",
      "children": "Push Me",
      "className": "Button-hYXUXp MvvlT",
      "disabled": false,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "warning",
        "children": "Push Me",
        "className": "Button-hYXUXp MvvlT",
        "disabled": false,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 9`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="link"
    disabled={false}
    renderAs="button"
    size="normal"
  >
    Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "link",
      "children": "Push Me",
      "className": "Button-hYXUXp dSZpHp",
      "disabled": false,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "link",
        "children": "Push Me",
        "className": "Button-hYXUXp dSZpHp",
        "disabled": false,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 10`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="link"
    disabled={true}
    renderAs="button"
    size="normal"
  >
    Don't Push Me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "link",
      "children": "Don't Push Me",
      "className": "Button-hYXUXp dSZpHp",
      "disabled": true,
      "renderAs": "button",
      "size": "normal",
    },
    "ref": null,
    "rendered": "Don't Push Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "link",
        "children": "Don't Push Me",
        "className": "Button-hYXUXp dSZpHp",
        "disabled": true,
        "renderAs": "button",
        "size": "normal",
      },
      "ref": null,
      "rendered": "Don't Push Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Button Component snapshots should match snapshot 11`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    appearance="primary"
    disabled={false}
    renderAs="span"
    size="normal"
  >
    I'm not a button
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appearance": "primary",
      "children": "I'm not a button",
      "className": "Button-hYXUXp fzSAgN",
      "disabled": false,
      "renderAs": "span",
      "size": "normal",
    },
    "ref": null,
    "rendered": "I'm not a button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appearance": "primary",
        "children": "I'm not a button",
        "className": "Button-hYXUXp fzSAgN",
        "disabled": false,
        "renderAs": "span",
        "size": "normal",
      },
      "ref": null,
      "rendered": "I'm not a button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
